src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3331,13 +3367,27 @@ private bool HasTopLevelNullabilityConversion(TypeSymbolWithAnnotations source,
                         Debug.Assert((object)methodOpt != null);
                         Debug.Assert(methodOpt.ParameterCount == 1);
                         var parameter = methodOpt.Parameters[0];
+                        var parameterType = parameter.Type;
+                        TypeSymbolWithAnnotations underlyingOperandType = default;
+                        bool isLiftedConversion =
+                            operandType.IsNullableType() &&
+                            !parameterType.IsNullableType() &&
+                            parameterType.Equals(underlyingOperandType = operandType.GetNullableUnderlyingType(), TypeCompareKind.AllIgnoreOptions);
```

Comment:
>parameterType.Equals(underlyingOperandType = operandType.GetNullableUnderlyingType(), TypeCompareKind.AllIgnoreOptions) [](start = 28, length = 119)

It feels like this condition is too strict. For example, it is valid for a lifted conversion to have ```long``` as the parameter type and ```int?``` as the operand type.

Reply:
The `operand -> conversion "from" type` conversion above should represent the conversion from `int?` to `long?`.

---
In reply to: [238872741](https://github.com/dotnet/roslyn/pull/31499#discussion_r238872741) [](ancestors = 238872741)

Reply:
Do we have a test for a scenario like this? Also, the fact that we are "redefining" what operandType represents throughout this sequence, makes it very hard to reason about the code. Consider introducing locals with specific names.

---
In reply to: [239170622](https://github.com/dotnet/roslyn/pull/31499#discussion_r239170622) [](ancestors = 239170622,238872741)

Reply:
Added tests.

---
In reply to: [239174936](https://github.com/dotnet/roslyn/pull/31499#discussion_r239174936) [](ancestors = 239174936,239170622,238872741)